extension/firefox/chrome/ShumwayCom.jsm

```diff
@@ -670,6 +686,34 @@ ShumwayChromeActions.prototype = {
     }
 
     return this.externalInterface.processAction(data);
+  },
+  
+  processFrame: function () {
+    if (this.embedTag.wrappedJSObject.processFrame) {
+      this.embedTag.wrappedJSObject.processFrame();
```

Comment:
It's unsafe to use wrappedJSObject for method calls. The embedTag (or even better, an object exposed on embedTag) shall expose methods that will pass callback functions to be invoked -- similar to setScreenShotCallback but for embedTag (instead of gfx frame). This way Xrays wrappers will be better utilized.